### DISCUSSION TITLE
Version stale when compiling

### DISCUSSION THREAD

**Discussion by grawlinson** [dani-garcia/vaultwarden#1806] [Q&A]
I'm one of the packagers for Vaultwarden on Arch Linux and I'm just updating the package, noticed this when compiling:

https://github.com/dani-garcia/vaultwarden/blob/72e1946ce5c3e6c33024be499b6e0e516a35ce92/Cargo.toml#L3

Seems like the version field in Cargo.toml does not get updated to reflect the tagged version.

**Comment by BlackDex** [comment]
It never does. We use the git tags for this.

**Reply by grawlinson** [reply]
That defeats the purpose of having Cargo.toml then, does it not? Surely it would be simple enough to bump the version when releasing.

**Reply by BlackDex** [reply]
The Cargo.toml is used for other things then just versioning.
Besides, we use the build.rs to modify this. And vaultwarden isn't a library which can be used for other rust projects.

Also, a lot of packages on Arch use either the git tags or hash to determine the pkg_version which should also work for this project i think.

**Reply by grawlinson** [reply]
Another method to install vaultwarden can be [cargo install](https://doc.rust-lang.org/cargo/commands/cargo-install.html), and because the project is on [crates.io](https://crates.io/crates/vaultwarden), it's extremely out of date.

**Reply by BlackDex** [reply]
Well the version at crates doesn't even represent the same version which is in Cargo.toml, so there seems to be something else there. Never had a project on crates, so not sure how that works.

**Reply by dani-garcia** [reply]
The version in crates.io is just a placeholder for now, we won’t be able to publish a real version there until we can compile on stable and all our dependencies are on crates themselves (so no git dependencies)
